packages/kit/src/runtime/server/respond.js

```diff
@@ -56,7 +56,22 @@ const allowed_page_methods = new Set(['GET', 'HEAD', 'OPTIONS']);
  */
 export async function respond(request, options, manifest, state) {
 	/** URL but stripped from the potential `/__data.json` suffix and its search param  */
-	const url = new URL(request.url);
+	const original_url = new URL(request.url);
+
+	// rewriteUrl could alter the given URL, so we pass a copy
+	let rewritten_url;
+	try {
+		rewritten_url = options.hooks.rewriteUrl({ url: new URL(original_url) });
+	} catch (e) {
+		return text('Internal Server Error', {
+			status: 500
+		});
```

Comment:
ideally this would go through the normal error handling mechanism, so that stuff gets logged and the user sees a designed error page

Reply:
Ah yeah, fair enough, and I see we're already returning plain text responses in other tricky cases. Returning nicer error pages would be a good enhancement down the line but it feels like a non-breaking change so we can file it under 'one day'